Remotion/Web/Development.WebTesting.TestSite.Shared/SmartPageErrorTestHelper.cs

```diff
@@ -0,0 +1,33 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+using System.Web;
+
+namespace Remotion.Web.Development.WebTesting.TestSite.Shared
+{
+  public static class SmartPageErrorTestHelper
+  {
+    public static void Execute (HttpRequest request, HttpResponse response)
+    {
+      if (request.Path.Contains("SmartPageErrorTest.wxe") && request.HttpMethod == "POST" && request.Params["TriggerAsyncResponseWithFullPageRenderingErrorButton"] != null)
+      {
+        response.StatusCode = int.Parse(request.Params["statusCode"]);
```

Comment:
here, please add a guard int.TryParse so empty string and incorrect data don't fail the request. maybe return a generic 400 result in that case (400 because it's a client error with incorrect data)